com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs

```diff
@@ -1091,6 +1090,9 @@ protected override void Render(ScriptableRenderContext renderContext, Camera[] c
 #endif
             Terrain.GetActiveTerrains(m_ActiveTerrains);
 
+            XRSystem.singlePassAllowed = m_Asset.currentPlatformRenderPipelineSettings.xrSettings.singlePass;
```

Comment:
cant we have the singlePassAllowed directly check for that value?

Reply:
I think we could once we migrate the setting to a common Core setting ? right now, `GlobalXRSettings` is from HDRP only but `XRSystem.singlePassAllowed` is in Core.